base/arrayshow.jl

```diff
@@ -371,7 +371,7 @@ preceded by `prefix`, supposed to encode the type of the elements.
 """
 function _show_nonempty(io::IO, X::AbstractMatrix, prefix::String)
     @assert !isempty(X)
-    limit = get(io, :limit, false)::Bool
+    limit::Bool = get(io, :limit, false)
```

Comment:
I would be surprised if this is really better --- this way it will have to do `convert(Bool, ::Any)` instead of just asserting the type.

Reply:
Here I was just being consistent with the surrounding code; someone went through previously and put assertions on all the `:limit` cases, and there were all of the `limit::Bool = ` style. The difference triggered just enough OCD that I changed it.

Reply:
Aha, then all of those should probably be changed the other way! :)